src/test/java/htsjdk/variant/variantcontext/writer/VCFWriterUnitTest.java

```diff
@@ -133,6 +137,49 @@ public void testBasicWriteAndRead(final String extension) throws IOException {
 
     }
 
+    /** test, using the writer and reader, that we can output and input a VCF body without problems */
+    @Test(dataProvider = "vcfExtensionsDataProvider")
+    public void testWriteAndReadVCFBody(final String extension) throws IOException {
+        final File fakeVCFFile = File.createTempFile("testWriteAndReadVCFBody.", extension);
+        fakeVCFFile.deleteOnExit();
+        if (".vcf.gz".equals(extension)) {
+            new File(fakeVCFFile.getAbsolutePath() + ".tbi").deleteOnExit();
+        } else {
+            Tribble.indexFile(fakeVCFFile).deleteOnExit();
+        }
+        metaData = new HashSet<VCFHeaderLine>();
+        additionalColumns = new HashSet<String>();
+        final SAMSequenceDictionary sequenceDict = createArtificialSequenceDictionary();
+        final VCFHeader header = createFakeHeader(metaData, additionalColumns, sequenceDict);
+        final VariantContextWriter writer = new VariantContextWriterBuilder()
+                .setOutputFile(fakeVCFFile)
+                .setReferenceDictionary(sequenceDict)
+                .setOptions(EnumSet.of(Options.ALLOW_MISSING_FIELDS_IN_HEADER, Options.INDEX_ON_THE_FLY))
+                .build();
+        writer.setVcfHeader(header);
+        writer.add(createVC(header));
+        writer.add(createVC(header));
+        writer.close();
+
+        final VCFCodec codec = new VCFCodec();
+        codec.setVCFHeader(header, VCFHeaderVersion.VCF4_0);
+        AsciiLineReaderIterator iterator;
+        if (".vcf.gz".equals(extension)) {
+            iterator = new AsciiLineReaderIterator(new AsciiLineReader(new BlockCompressedInputStream(fakeVCFFile)));
+        } else {
+            iterator = new AsciiLineReaderIterator(new AsciiLineReader(new FileInputStream(fakeVCFFile)));
```

Comment:
Also, you should just use `AbstractFeatureReader.getFeatureReader()` here to get a reader on the vcf file -- it will handle the block-gzipped case for you automatically.